### DISCUSSION TITLE
Show FileSystem view

### DISCUSSION THREAD

**Discussion by vHanda** [GitJournal/GitJournal#831] [Feature Requests]
It would be nice to have a view where both folders and files are shown together. And by files, I mean all files, not just the notes. The files which are not notes, can be greyed out, and we could even provide additional information about why that file was not loaded.

I'm not sure if this File System view should go in the sidebar or it should just be another view for any folder.

---
> Please vote on this issue with a 👍reaction. It's one of the primary mechanisms used to prioritize issues.

**Comment by chris2fr** [comment]
I am voting this up because I would like a filesystem view, but I also wish for a view of files before or after folders (or mixed) based on preferences. These are two different things. A file system view gives me access to everything whereas a mixed system view give access to the same.infomation available here. I think this is more a mixed-system view request.

**Comment by vHanda** [comment]
Btw, if you go to `Settings -> Experimental Features -> Show File System` you can try out what I have done so far, it's still in the very very early stages.

**Comment by chris2fr** [comment]
Nice start.

Noir yet usable, but good idea. Nothing happens when I click on the
different elements

**Comment by caracolmzd** [comment]
I want to down-vote this feature.

I think more design work needs to be put into how to manage attachments specifically, and not a general file explorer.

I'm disappointed that I can't find my GitJournal files using my android File Manager. I don't know about you iOS users... maybe a system utility isn't an option. Why re-create the wheel in-app?

Relative links in documents don't appear to work, I wonder if this is related to the underlying approach to files, and again would be resolved if a native approach were taken. Excuse me, I haven't even begun to try to understand the architecture of the app yet.

**Comment by vHanda** [comment]
I didn't think that allowing moving the git repo to a folder discoverable by another app would remove the need of implementing this on Android. This is another bug for that. I'll raise of it priority and try to get it done sooner.


Could you please open a new bug about the relative links not working. They should be. If I can reproduce it, i'll fix it ASAP.

**Comment by caracolmzd** [comment]
The need in the OP seems pretty well handled by any off the shelf file manager, so I don't understand the vision for this feature.

I created an issue for relative links: #256

**Comment by jlamoure** [comment]
For myself, I would like this filesystem view to be my primary mode of navigating my notes repo. At the moment Gitjournal seems to cater to tag-based information management, I manage with folders. Delegating this to an external file manager + enabling the "Store Repo Externally" doesn't make sense to me at all.

As is, I have to do a lot of scrolling to get where I want to go using the "Folders" view. One step up would be having these folders collapsed by default. But, I would strongly prefer a series of taps in a filesystem type view to tapping and scrolling with a "Folders" view with folders auto-collapsed.
